search/lib/cmr.js

```diff
@@ -32,293 +31,190 @@ const STAC_SEARCH_PARAMS_CONVERSION_MAP = {
  */
 async function findCollections (params = {}) {
-  const response = await cmrSearch('/collections.json', params);
-  return response.data.feed.entry;
-}
+  logger.debug(`findCollections [${JSON.stringify(scrubSensitive(params))}]`);
 
-/**
- * Search CMR for granules in UMM_JSON format
- */
-async function getGranulesUmmResponse(params = {}, opts = settings) {
-  if (opts.cmrStacRelativeRootUrl === '/cloudstac') {
-    return await cmrSearchPost('/granules.umm_json', params);
-  }
-  return await cmrSearch('/granules.umm_json', params);
+  const { data } = await cmrSearch('/collections.json', params);
+  return _.get(data, 'feed.entry', []);
 }
 
 /**
- * Search CMR for granules in JSON format.
+ * Fetch a concept from CMR by conceptId.
+ *
+ * @param {string} cmrConceptId
+ * @returns {Promise.<CmrConcept>}
  */
-async function getGranulesJsonResponse(params = {}, opts = settings) {
-  if (opts.cmrStacRelativeRootUrl === '/cloudstac') {
-    return await cmrSearchPost('/granules.json', params);
-  }
-  return await cmrSearch('/granules.json', params);
+async function fetchConcept (cmrConceptId, opts = {format: "json"}) {
+  logger.debug(`fetchConcept [${cmrConceptId}][${JSON.stringify(scrubSensitive(opts))}]`);
+
+  const { format } = opts;
+  const extension = format ? `.${format}` : "";
+
+  const { data } = await cmrSearch(`/concepts/${cmrConceptId}${extension}`);
+  return data;
 }
 
 /**
- * Merge a list of umm and json formatted granules into a list of STAC granules.
+ * @typedef CmrGranResult
+ * @property {string} hits
+ * @property {Array.<Granule>} granules
  */
-function buildStacGranules(jsonGranules = [], ummGranules = []) {
-  const granules = jsonGranules.reduce(
-    (obj, item) => ({
-      ...obj,
-      [item.id]: item
-    }),
-    {}
-  );
-  ummGranules.forEach((g) => {
-    granules[g.meta['concept-id']].meta = g.meta;
-    granules[g.meta['concept-id']].umm = g.umm;
-  });
 
-  return granules;
-}
 /**
  * Search CMR for granules matching CMR query parameters
+ *
  * @param {object} params Object of CMR Search parameters
+ * @returns {Promise<CmrGranResult>}
  */
-async function findGranules (params = {}) {
-  // TODO swap these for single request for STAC format
-  const jsonResponse = await getGranulesJsonResponse(params);
-  const ummResponse = await getGranulesUmmResponse(params);
-
-  const granules = buildStacGranules(jsonResponse.data.feed.entry,
-                                     ummResponse.data.items);
+async function findGranules(params = {}) {
+  const usePost = settings.cmrStacRelativeRootUrl === '/cloudstac';
+  logger.debug(`Searching CMR for granules ${JSON.stringify(params)}`);
+
+  const { data: gJson, headers } = usePost ?
+        await cmrSearchPost('/granules.json', params) :
+        await cmrSearch('/granules.json', params);
+
+  const { data: gUmm } = usePost ?
+        await cmrSearchPost('/granules.umm_json', params) :
+        await cmrSearch('/granules.umm_json', params);
```

Comment:
We continue to make the call to both UMM and JSON until we can convert to a single graphQL call.